test/logflare/gen_singleton_test.exs

```diff
@@ -77,4 +77,21 @@ defmodule Logflare.GenSingletonTest do
       end)
     end
   end
+
+  describe "Watcher child_spec/1" do
+    test "defaults restart to :permanent" do
```

Comment:
This kinda tests implementation, as opposed to starting up an instance of GenSingleton and then sending the exit signal